### ISSUE TITLE
`serving_client` returns empty tensor

### ISSUE THREAD

**Issue by finbarrtimbers** [mozilla/DeepSpeech#567]
I'm trying to get the model to work with Tensorflow Serving. I'm working with the commit where `serving_client` was introduced ([2740336d08](https://github.com/mozilla/DeepSpeech/tree/2740336d089b592f905e0236a686aa71096dd28f/serving_client)). I've got a trained version of the model, but when I try to make predictions using the `serving_client`, the results tensor is equal to 

    dtype: DT_INT64 tensor_shape { dim { size: 1 } dim { size: 1 } dim { size: 1 } } int64_val: 0 [[[0]]]

Is there anything I can do to get this working?

**Comment by kdavis-mozilla** [comment]
@Cwiiis I'd guess you'd be the expert here.

**Comment by Cwiiis** [comment]
The serving client is deprecated, but I'd recommend using a later commit, perhaps before it was removed, or the last commit that changed any of the files in that directory.

For this specific problem, I'd need more details to comment - How was the model trained and exported, what input is being used, does the model work as expected when called from the notebook, how was the serving client run, what's the terminal output, etc.

**Comment by finbarrtimbers** [comment]
Thanks. This is the last commit that changed any of the files in that directory (I believe).

I trained the model on the LDC93S1 dataset for 50 epochs and exported it using the default export mechanism. I'm trying to use the LDC93S1.wav file to make predictions. I used the same commit for everything.

I haven't tried the notebook, but I will. 

The terminal output is a blank row:

```
$ ./bazel-bin/deepspeech_client/deepspeech_client --server=localhost:9000 --file=/root/test.wav

$
```

**Comment by Cwiiis** [comment]
Sorry to ask the obvious questions, but you've started the server I assume? (as documented in the README) - what's the terminal output of the server?

**Comment by finbarrtimbers** [comment]
Yes, I have. When I run 

```
bazel run -c opt -- //tensorflow_serving/model_servers:tensorflow_model_server \
    --port=9000 \
    --model_name=deepspeech \
    --model_base_path=/model/path
```

replacing `model_base_path` with my path, the model seems to load normally, ending with this output:


```
I external/org_tensorflow/tensorflow/contrib/session_bundle/session_bundle.cc:135] Running restore op for SessionBundle: save/restore_all, save/Const:0
I external/org_tensorflow/tensorflow/contrib/session_bundle/session_bundle.cc:244] Loading SessionBundle: success. Took 1910202 microseconds.
I tensorflow_serving/core/loader_harness.cc:86] Successfully loaded servable version {name: deepspeech version: 1}
I tensorflow_serving/model_servers/main.cc:256] Running ModelServer at 0.0.0.0:9000 ...
```

The server output doesn't change when I run the client, but the client isn't able to run without the server running, so it does appear to be connecting to the server.

**Comment by Cwiiis** [comment]
It could just be that the output is an empty string, I'd recommend verifying behaviour using the ldc93s1 dataset to make sure things work as expected. My greater recommendation would be to not use the serving client though.

**Comment by finbarrtimbers** [comment]
Yeah, fair enough. Thanks for the advice, I'll play around with it.
